sdk/applicationinsights/ci.yml

```diff
@@ -0,0 +1,42 @@
+# DO NOT EDIT THIS FILE
+# This file is generated automatically and any changes will be lost.
+
+resources:
+  repositories:
+    - repository: azure-sdk-build-tools
+      type: git
+      name: internal/azure-sdk-build-tools
+    - repository: azure-sdk-tools
+      type: github
+      name: Azure/azure-sdk-tools
+      endpoint: azure
+
+trigger:
+  branches:
+    include:
+      - master
+      - feature/*
+      - release/*
+      - hotfix/*
+  paths:
+    include:
+      - sdk/applicationinsights/
+
+pr:
+  branches:
+    include:
+      - master
+      - feature/*
+      - release/*
+      - hotfix/*
+  paths:
+    include:
+      - sdk/applicationinsights/
+
+stages:
+  - template: ../../eng/pipelines/templates/stages/archetype-sdk-client.yml
+    parameters:
+      ServiceDirectory: applicationinsights
+      Artifacts:
+        - name: azure-applicationinsights-query
```

Comment:
Do you have a strategy for generating these other ci.yaml files as well as their artifact lists or are you just doing them manually?

Reply:
Doing them manually - for now.

There are a bunch of cases where it isn't clear whether a package should be published or not - so it might be just as much effort managing the exclusions. There have also been a bunch of cases where a module starts appearing in the source tree but isn't in a buildable/releasable state so it would be difficult to ignore "for a while" if this part was automated.

One of the storage crypto packages for Java was a good example of this - can't think of an example for JavaScript right now but I'm sure it'll come up.

One idea that I had was generating the root ci.yml file and then having a ci.config.yml file along-side it which contains minimal extra configuration. Tooling would then process the tree and make sure all the ci.yml files are consistent (all have the right repositories, filters/triggers etc) and then reference the ci.config.yml file as a template which would contain the artifact list and in turn reference its downstream template.

Reply:
>One idea that I had was generating the root ci.yml file and then having a ci.config.yml file along-side it which contains minimal extra configuration. Tooling would then process the tree and make sure all the ci.yml files are consistent (all have the right repositories, filters/triggers etc) and then reference the ci.config.yml file as a template which would contain the artifact list and in turn reference its downstream template.

That might be a reasonable approach. I really want to make sure we have an easy way to maintain all this duplicated stuff in the ci.yml files.

Reply:
Created an issue to track this: https://github.com/Azure/azure-sdk/issues/654